fahrschein/src/main/java/org/zalando/fahrschein/ReaderManager.java

```diff
@@ -0,0 +1,10 @@
+package org.zalando.fahrschein;
+
+import org.zalando.fahrschein.domain.Subscription;
+
+import java.util.Optional;
+
+@FunctionalInterface
+public interface ReaderManager {
```

Comment:
In your first approach the `Listener` interface exposed a method `continueReading`? Why did you separate it now from the `Listener`?